Externals/curl/CMakeLists.txt

```diff
@@ -1,62 +1,47 @@
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/curl/include)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/curl/lib)
-
-set(CURL_LIBS MbedTLS::mbedtls zlibstatic)
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    set(use_core_foundation ON)
-
-    find_library(SYSTEMCONFIGURATION_FRAMEWORK "SystemConfiguration")
-    if(NOT SYSTEMCONFIGURATION_FRAMEWORK)
-        message(FATAL_ERROR "SystemConfiguration framework not found")
-    endif()
-
-    list(APPEND CURL_LIBS "-framework SystemConfiguration")
+set(BUILD_CURL_EXE OFF)
+set(BUILD_EXAMPLES OFF)
+set(BUILD_LIBCURL_DOCS OFF)
+set(BUILD_MISC_DOCS OFF)
+set(BUILD_SHARED_LIBS OFF)
+set(BUILD_STATIC_LIBS ON)
+set(BUILD_TESTING OFF)
+set(CURL_ENABLE_EXPORT_TARGET OFF)
+
+set(HTTP_ONLY ON)
+
+set(CURL_USE_LIBPSL OFF)
+set(CURL_USE_LIBSSH2 OFF)
+set(CURL_ZLIB OFF CACHE BOOL "" FORCE)
+set(CURL_ZSTD OFF)
+set(USE_LIBIDN2 OFF)
+set(USE_NGHTTP2 OFF)
+
+if(UNIX)
+  # We use mbedtls on Unix(-like) systems and Android.
+  set(CURL_USE_OPENSSL OFF)
+
+  if(MBEDTLS_FOUND)
```

Comment:
`add_subdirectory()` with a CMakeLists.txt that wasn't written to account for that scenario is always a bit iffy, so this is probably fine, but for the record... The `{LIBRARY}_FOUND` variables are only set if anyone has previously invoked `find_package()` or similar. Since we do search for MBEDTLS unconditionally in the main CMakeLists.txt that should work fine, but it might be worth noting where this comes from.

Reply:
Added a comment.